de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/navigation/NavigationPresenterImpl.java

```diff
@@ -244,7 +339,69 @@ public void onImportFromUrlSelected(final ImportFromUrlSelected event) {
         if (destinationFolder == null) {
             destinationFolder = getSelectedUploadFolder();
         }
-        eventBus.fireEvent(new RequestImportFromUrlEvent(destinationFolder));
+
+        if (canUpload(destinationFolder)) {
+            Folder finalDestinationFolder = destinationFolder;
+            urlImportDlgProvider.get(new AsyncCallback<FileUploadByUrlDialog>() {
+                @Override
+                public void onFailure(Throwable caught) {}
+
+                @Override
+                public void onSuccess(FileUploadByUrlDialog dialog) {
+                    dialog.show(finalDestinationFolder);
+                    dialog.addCancelButtonSelectHandler(selectEvent -> dialog.hide());
+                    dialog.addOkButtonSelectHandler(selectEvent -> {
+                        dialog.getOkButton().setEnabled(false);
+                        handleUrlImport(finalDestinationFolder, dialog.getFieldToStatusMap(), dialog);
+                    });
+                }
+            });
+        }
+    }
+
+    void handleUrlImport(Folder destinationFolder, Map<Field<String>, Status> fieldToStatusMap, FileUploadByUrlDialog dialog) {
+        final FastMap<Field<String>> destResourceMap = new FastMap<>();
+
+        for (Map.Entry<Field<String>, Status> entry : fieldToStatusMap.entrySet()) {
+            Field<String> field = entry.getKey();
```

Comment:
we can just use `map.keySet()` instead of `map.entrySet()` if we want to use just the key.